File: arrays/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result;
        unordered_map<string,vector<int>> map = {};
        
        // fill map
        for(int i=0; i<strs.size(); i++){
            string tempStr = strs[i];
            sort(tempStr.begin(), tempStr.end());
            map[tempStr].push_back(i);
        }
        
        // iterate through map and fill result
        for(auto i: map){
            vector<string> temp;
            for(auto j: i.second){
                temp.push_back(strs[j]);
            }
            result.push_back(temp);  
        }
               
        return result;
    }
};

File: arrays/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        map = {}
        for s in strs:
            sorted_s = ''.join(sorted(s))
            if sorted_s in map:
                map[sorted_s].append(s)
            else:
                map[sorted_s] = [s]
        
        result = []
        for k in map.keys():
            result.append(map[k])
        
        return result
